src/pages/settings/Subscription/SubscriptionDetails/index.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import Icon from '@components/Icon';
+import * as Illustrations from '@components/Icon/Illustrations';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import type {OptionsPickerItem} from '@components/OptionsPicker';
+import OptionsPicker from '@components/OptionsPicker';
+import Section from '@components/Section';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function SubscriptionDetails() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [privateSubscription] = useOnyx(ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION);
+
+    const [selectedOption, setSelectedOption] = React.useState<string>(privateSubscription?.type ?? CONST.SUBSCRIPTION.TYPE.ANNUAL);
```

Comment:
```suggestion
    const [selectedOption, setSelectedOption] = React.useState(privateSubscription?.type ?? CONST.SUBSCRIPTION.TYPE.ANNUAL);
```

I think we don't need to specify the type

Reply:
I removed it but I had to add type casting to `onOptionSelected`, let me know if that's ok

Reply:
@JKobrynski I think you can try to get rid of type casting if you update `OpenPicker` typing to use generics.
For example
```
type OptionsPickerItem<TKey extends string> = {
    /** A unique identifier for each option */
    key: TKey;

    /** Text to be displayed */
    title: TranslationPaths;

    /** Icon to be displayed above the title */
    icon: IconAsset;
};

type OptionsPickerProps<TKey  extends string> = {
    /** Options list */
    options: Array<OptionsPickerItem<TKey>>;

    /** Selected option's identifier */
    selectedOption: TKey;

    /** Option select handler */
    onOptionSelected: (option: TKey) => void;

    /** Indicates whether the picker is disabled */
    isDisabled?: boolean;

    /** Optional style */
    style?: StyleProp<ViewStyle>;
};

function OptionsPicker<TKey extends string>({options, selectedOption, onOptionSelected, style, isDisabled}: OptionsPickerProps<TKey>) {
...
```

Reply:
@VickyStash works perfectly, thanks!